src/main/resources/drivers/CiscoNext-GenSeries.js

```diff
@@ -0,0 +1,478 @@
+    response: "%password%"
+  },
+
+  anykey: {
+    // Some images show "Press any key..." prior to the login prompts.
+    prompt: /Press any key to continue\.{0,3}\s*$/mi,
+    response: "\n",
+    optional: true
+  },
+
+  /* -------------------------- Normal modes --------------------------- */
+
+  exec: {
+    prompt: PROMPT_EXEC,
+    pager: {
+      match: PAGER_RE,
+      response: " "
+    }
+  },
+
+  enable: {
+    prompt: PROMPT_ENABLE,
+    // Run once we’re at an enable-capable prompt.
+    commands: [
+      // Some images don’t require enable; keep optional.
+      { command: "enable", expect: /Password:\s*$/mi, response: "%enablePassword%", optional: true },
+      { command: "terminal datadump", optional: true },
+      { command: "terminal length 0", optional: true },
+      { command: "terminal width 511", optional: true }
+    ],
+    pager: {
+      match: PAGER_RE,
+      response: " "
+    }
+  }
+};
+
+/* ========================= HELPERS ========================= */
+
+/**
+ * Normalize an interface object for Netshot without adding unsupported fields (e.g., VRF).
+ */
+const ensureIfaceShape = (obj, fallbackName, isL3) => {
+  obj = (obj && typeof obj === "object") ? obj : {};
+  obj.name = (obj.name != null ? String(obj.name) : String(fallbackName || "")).trim();
+  if (!obj.name) return null;
+
+  if (obj.description == null) obj.description = "";
+  if (obj.level3 == null) obj.level3 = !!isL3;
+
+  if (obj.enabled == null) obj.enabled = false;
+  if (obj.adminStatus == null) obj.adminStatus = obj.enabled ? "Up" : "Down";
+  if (obj.linkStatus == null) obj.linkStatus = obj.adminStatus;
+
+  if (!Array.isArray(obj.ip)) obj.ip = [];
+  obj.ip = obj.ip.filter(x => x && typeof x === "object" && (x.ip || x.address));
+
+  // Drop empty MAC keys; keep whatever delimiter style the device prints (dashes/colons/dots)
+  if (!obj.mac) delete obj.mac;
+  if (!obj.macAddress) delete obj.macAddress;
+
+  return obj;
+};
+
+const scrubMacs = o => {
+  if (!o) return o;
+  if (!o.mac) delete o.mac;
+  if (!o.macAddress) delete o.macAddress;
+  return o;
+};
+
+// Dedup guards
+const addedIfaces = {};
+const addIfaceOnce = (device, kindObj) => {
+  if (!kindObj || !kindObj.name) return;
+  const key = String(kindObj.name).toLowerCase();
+  if (addedIfaces[key]) return;
+  addedIfaces[key] = true;
+  device.add("networkInterface", kindObj);
+};
+
+const addedModules = {};
+const addModuleOnce = (device, mod) => {
+  if (!mod) return;
+  // Build a stable dedup key: slot|partNumber|serialNumber (lower-cased)
+  const k = [
+    mod.slot || "",
+    mod.partNumber || "",
+    mod.serialNumber || ""
+  ].join("|").toLowerCase();
+  if (addedModules[k]) return;
+  addedModules[k] = true;
+  device.add("module", mod);
+};
+
+/* ========================= SNAPSHOT ========================= */
+
+function snapshot(cli, device, config) {
+  // Land in enable mode first.
+  try { cli.macro("enable"); } catch (e) {}
```

Comment:
An error from `cli.macro` should be propagated (no catch) to snapshot task it fails there before attempting to issue commands in an unknown context.